hpy/devel/include/cpython/hpyfunc_trampolines.h

```diff
@@ -69,4 +69,17 @@
         return _h2py(IMPL(_HPyGetContext(), _py2h(self), _py2h(obj), op)); \
     }
 
+/* With the cpython ABI, the casts from Py_buffer* to HPy_buffer*
+ * happen to Just Work, which makes things easier. */
+#define _HPyFunc_TRAMPOLINE_HPyFunc_GETBUFFERPROC(SYM, IMPL) \
+    static int SYM(PyObject *arg0, Py_buffer *arg1, int arg2) \
+    { \
+        return (IMPL(_HPyGetContext(), _py2h(arg0), (HPy_buffer*)arg1, arg2)); \
```

Comment:
I *think* now this cast to `HPy_buffer` is wrong? `arg1` should be a `cpy_Py_buffer` which is just `Py_buffer` right? Maybe now the cast can just be removed?

Reply:
yes but `IMPL` takes an `HPy_buffer`, so the cast is needed, I think. The actual problem is that casting `Py_Buffer*` to `HPy_Buffer*` is probably undefined behavior, although it will work in practice.  We should put a comment and explain the issue.

Reply:
Does this mean that a type that registers a getbufferproc slot using HPy but is called with `PyObject_GetBuffer` will return a buffer object where `obj` is an `HPy` and will possibly break when `PyBuffer_Release` is called on and attempts to `Py_XDECREF` `obj`?

Reply:
Unless we change the implementation of `HPy`, the likely issue is that `Py_buffer` and `HPy_buffer` could have different layouts, so any member could be wrong and we'd quickly end up with a segfault as soon as the code does anything with the buffer.